assets/js/search.js

```diff
@@ -0,0 +1,124 @@
+(function() {
+    'use stict';
+    var activityCards, find, mapCardToClassName, mapCardToHeading, hideCard,
+        showCard, filterCards, onSeachChange, onClearSearch, createSearchInput,
+        activityContainer, searchInput;
+
+    activityCards = [].slice.call(document.querySelectorAll('.activity-card'));
+
+    find = function(needle, haystack) {
+        var regex;
+
+        regex = new RegExp(needle);
+        return regex.test(haystack);
+    };
+
+    mapCardToClassName = function(card) {
+        return card.className;
+    };
+
+    mapCardToHeading = function(card) {
+        var heading;
+
+        heading = card.querySelector('h2').textContent;
+        return heading.toLowerCase();
+    };
+
+    hideCard = function(card) {
+        if (card.className.split(' ').indexOf('fade') === -1) {
+            card.className += ' fade';
+        }
+        // Bootstrap's .fade animates for 150 ms
+        setTimeout(function() { card.style.display = 'none'; }, 155);
+    };
+
+    showCard = function(card) {
+        // kudos https://gist.github.com/k-gun/c2ea7c49edf7b757fe9561ba37cb19ca
+        // resp. https://developer.mozilla.org/en-US/docs/Web/API/Element/classList#Polyfill
+        if (card.className.split(' ').indexOf('fade') > -1) {
+            card.className = card.className.replace(new RegExp('(^| )fade( |$)'), '');
+        }
+        // In order to kick in after async hideCard calls
+        // 300 ms is a magic limit, where effect becomes notable to humans
+        setTimeout(function() { card.style.display = ''; }, 300);
```

Comment:
I don't know, which `display` was assigned by CSS files. The inline `style` attribute wins (because specificity). By setting it to an empty string, the inline CSS rule becomes invalid and will be removed by the browser. The CSS `display` kicks in again.